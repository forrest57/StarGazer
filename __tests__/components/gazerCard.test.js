import GazerCard from '../../components/GazerCard'
import React from 'react'
import renderer from 'react-test-renderer'

const prop = 'lorem'
const imgUri = 'https://placekitten.com/200/300'
describe('<GazerCard />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<GazerCard login={prop} avatar={imgUri} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('renders passed props', () => {
    const gazerComponent = renderer
      .create(<GazerCard login={prop} avatar={imgUri} />)
      .toJSON()
    const renderedPropText = gazerComponent.children[1].children[0]
    const renderedPropUri = gazerComponent.children[0].props.source.uri
    expect(renderedPropText).toBe(prop)
    expect(renderedPropUri).toBe(imgUri)
  })
})
