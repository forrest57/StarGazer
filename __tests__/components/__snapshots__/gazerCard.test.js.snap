// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GazerCard /> renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#374151",
      "borderRadius": 6,
      "display": "flex",
      "flexDirection": "row",
      "marginBottom": 8,
      "marginTop": 8,
      "paddingBottom": 16,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 16,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://placekitten.com/200/300",
      }
    }
    style={
      Object {
        "aspectRatio": 1,
        "borderRadius": 9999,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "gray",
        "fontSize": 22,
        "marginLeft": 15,
      }
    }
  >
    lorem
  </Text>
</View>
`;

exports[`<gazerCard /> renders at all 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#374151",
      "borderRadius": 6,
      "display": "flex",
      "flexDirection": "row",
      "marginBottom": 8,
      "marginTop": 8,
      "paddingBottom": 16,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 16,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://placekitten.com/200/300",
      }
    }
    style={
      Object {
        "aspectRatio": 1,
        "borderRadius": 9999,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "gray",
        "fontSize": 22,
        "marginLeft": 15,
      }
    }
  >
    lorem
  </Text>
</View>
`;

exports[`<gazerCard /> renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#374151",
      "borderRadius": 6,
      "display": "flex",
      "flexDirection": "row",
      "marginBottom": 8,
      "marginTop": 8,
      "paddingBottom": 16,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 16,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://placekitten.com/200/300",
      }
    }
    style={
      Object {
        "aspectRatio": 1,
        "borderRadius": 9999,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "gray",
        "fontSize": 22,
        "marginLeft": 15,
      }
    }
  >
    lorem
  </Text>
</View>
`;
